### PR TITLE
Allow to use SwipeItem only with text or icon

### LINKED ISSUE #8777: [Bug] SwipeView crash if Text not is set on SwipeItem
### Steps to Reproduce

1. Create a ColletionView
2. Add SwipeItems to the template
3. Add an IconImageSource to the SwipeItems
4. Swipe

### Expected Behavior
Only icons visible for SwipeItem

### Actual Behavior
When running it on iOS 13.2.2 simulator NullReferenceException when the user swipes.

### Basic Information

- Version with issue: Xamarin.Forms 4.4.0.001220-pre3

### Reproduction repo
Left items only have an icon in this sample and it crashes. Right items are working.

https://github.com/dhindrik/SwipeViewIconCrash

### LINKED ISSUE #9108: iOS SwipeViewRenderer NullReferenceException When SwipeItem.Text is Null
### Description

On iOS, when `SwipeItem.Text == null` a NullReferenceException is thrown when the SwipeItem is revealed by the user.

### Steps to Reproduce

1. Download/clone this reproduction sample: https://github.com/brminnick/SwipeViewCrashRepro
2. In Visual Studio, open `SwipeRendererRepro.sln`
3. In Visual Studio, build/deploy `SwipeRendererRepro.iOS` to an iOS Simulator or Device
4. On the iOS app, on the yellow `Label`, swipe left-to-right
5. On the iOS app, confirm the `SwipeItem` is working properly 
6. On the iOS app, click the green `Test` text to dismiss the `SwipeItem`
7. On the iOS app, on the yellow `Label`, swipe right-to-left
8. Confirm the app crashes

### Expected Behavior

Swiping right-to-left should reveal a `SwipeItem` with `BackgroundColor = Color.Red`

### Actual Behavior

Swiping right-to-left throws a NullReferenceException

### Basic Information

- Version with issue: v4.4.0.991265
- Last known good version: Unknown
- IDE: VS for Mac
- Platform Target Frameworks: 
  - iOS:  13.6.0.12

### Reproduction Link

https://github.com/brminnick/SwipeViewCrashRepro

### Environment

=== Visual Studio Enterprise 2019 for Mac ===

Version 8.3.11 (build 1)
Installation UUID: d03a9496-0fde-4bad-85bd-1ff8c31cf324
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.16.1.24 (d16-3 / 08809f5b)

	Package version: 604000208

=== Mono Framework MDK ===

Runtime:
	Mono 6.4.0.208 (2019-06/07c23f2ca43) (64-bit)
	Package version: 604000208

=== NuGet ===

Version: 5.3.0.6192

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/3.1.100/Sdks
SDK Versions:
	3.1.100
	3.0.101
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.1.0
	3.0.1
	2.1.14

=== Xamarin.Profiler ===

Version: 1.6.12.29
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 11.3 (15712)
Build 11C29

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 13.6.0.12 (Visual Studio Enterprise)
Hash: e3c2b406d
Branch: xcode11.2
Build date: 2019-11-01 00:12:08-0400

=== Xamarin Designer ===

Version: 16.3.0.256
Hash: 8a223bfd7
Branch: remotes/origin/d16-3
Build date: 2019-11-01 21:02:02 UTC

=== Xamarin.Android ===

Version: 10.0.6.2 (Visual Studio Enterprise)
Commit: xamarin-android/d16-3/c407838
Android SDK: /Users/bramin/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 29.0.5
SDK Build Tools Version: 29.0.2

Build Information: 
Mono: mono/mono/2019-06@476d72b9e32
Java.Interop: xamarin/java.interop/d16-3@5836f58
LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
LibZip: nih-at/libzip/rel-1-5-1@b95cf3fd
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/bramin/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 1.4.0.65
Hash: c33b107
Branch: remotes/origin/d16-3
Build date: 2019-11-19 20:33:22 UTC

=== Android Device Manager ===

Version: 1.2.0.116
Hash: d2b2af0
Branch: remotes/origin/d16-3
Build date: 2019-11-19 20:33:42 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 803110001
Git revision: 6ee6ad2ec46ae5a08a1999ee4c815ac656a35b91
Build date: 2019-12-05 16:09:27+00
Build branch: release-8.3
Xamarin extensions: 56bd70ef2e327f71c615cfc29a47fd50685fadcb

=== Operating System ===

Mac OS X 10.15.2
Darwin 19.2.0 Darwin Kernel Version 19.2.0
    Sat Nov  9 03:47:04 PST 2019
    root:xnu-6153.61.1~20/RELEASE_X86_64 x86_64

### LINKED ISSUE #9271: SwipeItem Crash in iOS when Text = null
### Description
When the **Text** is not set in a **SwipteItem** and image bound to a **FontImageSource**, iOS will crash with a null exception.

Simply adding a **Text=" "** will work as a workaround

### Steps to Reproduce
    <SwipeItem      IconImageSource="{StaticResource TrashCanImage}"
                       BackgroundColor="{DynamicResource DeleteButtonColor}" />

**Using**

      <OnPlatform x:Key="ImageFontAwesomeFontFamily" x:TypeArguments="x:String"
                Android="fontawesome.ttf#Font Awesome 5 Pro"
                iOS="FontAwesome5Pro-Regular" />

       <FontImageSource 
         x:Key="TrashCanImage" 
         Size="34"
         FontFamily="{StaticResource ImageFontAwesomeFontFamily}"
         Glyph="{x:Static local:FontAwesomeFont.Trash}"/> 

### Expected Behavior
Works
### Actual Behavior
Crashes
### Basic Information
`"  at UIKit.UIStringDrawing.StringSize (Foundation.NSString This, UIKit.UIFont font) [0x00068] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/UIStringDrawing.g.cs:230 \n  at UIKit.UIStringDrawing.StringSize (System.String This, UIKit.UIFont font) [0x00007] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/UIKit/UIStringDrawing.cs:65 \n  at Xamarin.Forms.Platform.iOS.SwipeViewRenderer.UpdateSwipeItemInsets (UIKit.UIButton button, System.Single spacing) [0x00067] in D:\\a\\1\\s\\Xamarin.Forms.Platform.iOS\\Renderers\\SwipeViewRenderer.cs:393 \n  at Xamarin.Forms.Platform.iOS.SwipeViewRenderer.CreateSwipeItem (Xamarin.Forms.SwipeItem formsSwipeItem) [0x00051] in D:\\a\\1\\s\\Xamarin.Forms.Platform.iOS\\Renderers\\SwipeViewRenderer.cs:354 \n  at Xamarin.Forms.Platform.iOS.SwipeViewRenderer.UpdateSwipeItems () [0x00103] in D:\\a\\1\\s\\Xamarin.Forms.Platform.iOS\\Renderers\\SwipeViewRenderer.cs:311 \n  at Xamarin.Forms.Platform.iOS.SwipeViewRenderer.ProcessTouchMove (CoreGraphics.CGPoint point) [0x00083] in D:\\a\\1\\s\\Xamarin.Forms.Platform.iOS\\Renderers\\SwipeViewRenderer.cs:511 \n  at Xamarin.Forms.Platform.iOS.SwipeViewRenderer.HandleTouchInteractions (Xamarin.Forms.GestureStatus status, CoreGraphics.CGPoint point) [0x00021] in D:\\a\\1\\s\\Xamarin.Forms.Platform.iOS\\Renderers\\SwipeViewRenderer.cs:469 \n  at Xamarin.Forms.Platform.iOS.SwipeViewRenderer.HandleTouchInteractions (Foundation.NSSet touches, Xamarin.Forms.GestureStatus gestureStatus) [0x0002a] in D:\\a\\1\\s\\Xamarin.Forms.Platform.iOS\\Renderers\\SwipeViewRenderer.cs:458 \n  at Xamarin.Forms.Platform.iOS.SwipeViewRenderer.TouchesMoved (Foundation.NSSet touches, UIKit.UIEvent evt) [0x00000] in D:\\a\\1\\s\\Xamarin.Forms.Platform.iOS\\Renderers\\SwipeViewRenderer.cs:208 \n  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)\n  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/UIKit/UIApplication.cs:86 \n  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/UIKit/UIApplication.cs:65 \n  at Max.iOS.Application.Main (System.String[] args) [0x00001] in C:\\x\\Mobile\\Max.iOS\\Main.cs:14 "`

- Version with issue:
  - iOS:  Newest
 

 
### Workaround
Add a blank to Text  (Text=" ")

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Allow to use SwipeItem only with text or icon.

### Issues Resolved ### 

- fixes #8777
- fixes #9108
- fixes #9271

### API Changes ###
 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android

### Behavioral/Visual Changes ###

None

#### Before
`System.NullReferenceException` has been thrown on iOS, no centered icon on Android.

#### After
issue8777-droid
issue8777-ios

### Testing Procedure ###
Launch Core Gallery and navigate to the issue 8777. Open the SwipeView and verify that only appears the centered icon.

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue8777.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+using NUnit.Framework;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+#if UITEST
+	[Category(UITestCategories.SwipeView)]
+#endif
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 8777, "SwipeView crash if Text not is set on SwipeItem", PlatformAffected.iOS)]
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue8777.cs]:
```suggestion
	[Issue(IssueTracker.Github, 8777, "SwipeView crash if Text not is set on SwipeItem", PlatformAffected.iOS | PlatformAffected.Android)]
```

Comment (jsuarezruiz) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue8777.cs]: Changed!

**Comment by samhouts** [comment]
All tests passed! Resolving conflicts...

**Comment by samhouts** [comment]
@jsuarezruiz Would you please resolve the conflict? Thanks!!

**Pull request merged**